### PR TITLE
Modify ExceptionHandler to support getHttpStatus

### PR DESCRIPTION
**Pull request by elazar**

### CODE REVIEW

File: src/Handler/ExceptionHandler.php [hunk 1]
```diff
@@ -71,7 +71,10 @@ public function __invoke(
             $response = $response->withHeader('Content-Type', $type);
 
             try {
-                $response = $response->withStatus($e->getCode());
+                $code = method_exists($e, 'getHttpStatus')
+                    ? $e->getHttpStatus()
+                    : $e->getCode();
```
Comment (shadowhand) [inline src/Handler/ExceptionHandler.php]: I'd rather see this handled with an if/else condition.

Comment (elazar) [inline src/Handler/ExceptionHandler.php]: Done.

**Comment by ameech** [comment]
:+1:

**Comment by elazar** [comment]
@shadowhand @ameech Bah, need a re - :+1: because I had to push a change to address @shadowhand's issue.

**Comment by ameech** [comment]
:+1:

File: tests/Handler/ExceptionHandlerTest.php [hunk 2]
```diff
@@ -82,3 +93,11 @@ public function testMethodNotAllowed()
         $this->assertEquals('GET,PUT', $response->getHeaderLine('Allow'));
     }
 }
+
+class ExceptionWithHttpStatusCode extends \Exception
```
Comment (shadowhand) [inline tests/Handler/ExceptionHandlerTest.php]: I would put this in a separate class. Having it here violates the requirement of "one class per file".

Comment (elazar) [inline tests/Handler/ExceptionHandlerTest.php]: Normally, I'd be inclined to agree, but I realistically can't see using this class anywhere else. I'd use a mock except that PHPUnit's mocking doesn't allow stubbing methods that don't exist in the parent class.

Comment (shadowhand) [inline tests/Handler/ExceptionHandlerTest.php]: Right I'd just add it alongside this class.

**Pull request merged**